agent/snmpvars.hpp

```diff
@@ -0,0 +1,121 @@
+/**
+ * @brief netsnmp_variable_list C++ wrapper.
+ *
+ * This file is part of phosphor-snmp project.
+ *
+ * Copyright (c) 2018 YADRO
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#pragma once
+
+namespace phosphor
+{
+namespace snmp
+{
+namespace agent
+{
+
+/**
+ * @brief SNMP representation of boolean type.
+ */
+enum class TruthValue : int
+{
+    True = 1,
+    False = 2,
+};
+
+struct VariableList
+{
+    /**
+     * @brief Fill snmp field with string value.
+     */
+    static void set(netsnmp_variable_list* var, const std::string& value)
+    {
+        snmp_set_var_typed_value(var, ASN_OCTET_STR, value.c_str(),
+                                 value.length());
+    }
+
+    /**
+     * @brief Fill snmp field with boolean value.
+     */
+    static void set(netsnmp_variable_list* var, bool value)
+    {
+        snmp_set_var_typed_integer(
+            var, ASN_INTEGER,
+            static_cast<int>(value ? TruthValue::True : TruthValue::False));
```

Comment:
Please replace this cast with something like an inline function or a macro (e.g. `SNMPBOOL(value)` or `toSNMPbool(value)`.

Reply:
done